### ISSUE TITLE
RStudio hang after reading in a not so big dataset

### ISSUE THREAD

**Issue by lvalnegri** [rstudio/rstudio#14531]
### System details

    RStudio Edition : Server
    RStudio Version : 2023.12.1 Build 402
    OS Version      : Ubuntu 22.04.4 LTS
    R Version       : R version 4.3.3 (2024-02-29) (svn rev 86002)

### Steps to reproduce the problem
```
options(timeout=120) # you possibly need to raise this value
tmpf <- tempfile()
download.file("https://dati.anticorruzione.it/opendata/download/dataset/partecipanti/filesystem/partecipanti_csv.zip", tmpf)
tmpd <- tempdir()
unzip(tmpf, exdir = tmpd)
y <- fread(file.path(tmpd, 'partecipanti_csv.csv'), na.strings = '')
ts <- unique(y[, .(tipo_soggetto)])[order(tipo_soggetto)][, id := 1:.N]
# anything you do afterwards is just lost
```

### Describe the problem in detail
After reading in the file using `data.table::fread`, RStudio just hang and it's not more capable to do anything, showing after a while the following:
```
Error: Unable to establish connection with session on RStudio Server. Please try logging in again in a new tab, then return to resume your session
```
But opening a new tab for a new login does not work, as that tab won't ever open.
Notice that Rstudio does not actually crash.
If on the other hand I work in the terminal, R works as expected.
Moreover, it's not a machine problem, I have 192GB RAM with 32 cores CPU.

### Describe the behavior you expected
Just working as it should?

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@lvalnegri Thank you for raising the issue! I've tried to reproduce it, but I'm able to load the data and continue using RStudio. What version of `data.table` are you using, so that I could double-check that?

@gtritchie Any suggestions of what to look for?

**Comment by lvalnegri** [comment]
> @lvalnegri Thank you for raising the issue! I've tried to reproduce it, but I'm able to load the data and continue using RStudio. What version of `data.table` are you using, so that I could double-check that?
data.table 1.15.2. 

But weirdly it has worked once since I posted, so I could do my job in that time, and it did the same just now on some other very small table. Now I'm more prone to think it's something weird going on with my machine (memory fault?), but I can't restart it until friday night anyway, it's a bare-metal a few 1Ks km from here used by multiple people, I've already been in a missed reboot mid-week, not pleasant! 
Anyway, I'm closing this now and in case reopen. 
Thanks!

> Note: I had to download the file in the terminal instead of via R because of a timeout, even after setting the max to 10 minutes.
oh wow, think that I had to download from that same site a 2GB NON-ZIPPED json (who on earth does that?)  and it took to me 1h22m! at one point I put 7200 for timeout just to be sure it won't stop again.
